src/main/java/net/openhft/chronicle/queue/impl/single/StoreAppender.java

```diff
@@ -856,12 +859,17 @@ public void close(boolean unlock) {
         }
 
         private void doRollback() {
-            // zero out all contents...
-            for (long i = positionOfHeader; i <= wire.bytes().writePosition(); i++)
-                wire.bytes().writeByte(i, (byte) 0);
-            long lastPosition = StoreAppender.this.lastPosition;
-            position0(lastPosition, lastPosition);
-            ((AbstractWire) wire).forceNotInsideHeader();
+            if (buffered) {
+                assert wire != StoreAppender.this.wire;
+                wire.clear();
+            } else {
+                // zero out all contents...
+                for (long i = positionOfHeader; i <= wire.bytes().writePosition(); i++)
+                    wire.bytes().writeByte(i, (byte) 0);
+                long lastPosition = StoreAppender.this.lastPosition;
+                position0(lastPosition, lastPosition);
+                ((AbstractWire) wire).forceNotInsideHeader();
+            }
```

Comment:
This isn't pretty, but it gets the job done. Previously `doRollback` assumed `StoreAppenderContext.wire == StoreAppender.wire`